File: lcy8047/13_RomantoInteger.cpp
class Solution {
public:
    int romanToInt(string s) {
        char prev_char = 0;
        int result = 0;
        for ( auto c : s )
        {
            switch ( c )
            {
                case 'I':
                {
                    result += 1;
                    break;
                }
                case 'V':
                {
                    if ( prev_char == 'I' )
                    {
                        result += 3;
                    }
                    else
                    {
                        result += 5;
                    }
                    break;
                }
                case 'X':
                {
                    if ( prev_char == 'I' )
                    {
                        result += 8;
                    }
                    else
                    {
                        result += 10;
                    }
                    break;
                }
                case 'L':
                {
                    if ( prev_char == 'X' )
                    {
                        result += 30;
                    }
                    else
                    {
                        result += 50;
                    }
                    break;
                }
                case 'C':
                {
                    if ( prev_char == 'X' )
                    {
                        result += 80;
                    }
                    else
                    {
                        result += 100;
                    }
                    break;
                }
                case 'D':
                {
                    if ( prev_char == 'C' )
                    {
                        result += 300;
                    }
                    else
                    {
                        result += 500;
                    }
                    break;
                }
                case 'M':
                {
                    if ( prev_char == 'C' )
                    {
                        result += 800;
                    }
                    else
                    {
                        result += 1000;
                    }
                    break;
                }
                default:
                {
                    assert( false );
                    break;
                }
            }
            prev_char = c;
        }
        return result;
    }
};

File: EunjiShin/LeetCode/13.roman-to-integer.java
package Easy;

/*
 * @lc app=leetcode id=13 lang=java
 *
 * [13] Roman to Integer
 */

// @lc code=start
class Solution {

    public int romanToInt(String s) {
        int result = 0;
        int n = s.length();
        int[] romanArray = makeRomanArray(s, n);

        for (int i = 0; i < n - 1; i++) {
            if (romanArray[i] < romanArray[i + 1])
                result -= romanArray[i];
            else
                result += romanArray[i];
        }
        return result + romanArray[n - 1];
    }

    private static int[] makeRomanArray(String s, int n) {
        int[] array = new int[n];
        for (int i = 0; i < n; i++) {
            switch (s.charAt(i)) {
                case 'I':
                    array[i] = 1;
                    break;
                case 'V':
                    array[i] = 5;
                    break;
                case 'X':
                    array[i] = 10;
                    break;
                case 'L':
                    array[i] = 50;
                    break;
                case 'C':
                    array[i] = 100;
                    break;
                case 'D':
                    array[i] = 500;
                    break;
                case 'M':
                    array[i] = 1000;
                    break;
            }
        }
        return array;
    }

}
// @lc code=end
